godot-macros/src/docs/markdown_converter.rs

```diff
@@ -5,91 +5,330 @@
+                    .filter_map(|child| self.walk_node(child, level))
+                    .collect();
 
-        Strong(strong) => format!("[b]{}[/b]", walk_nodes(&strong.children, definitions, "")),
+                join_if_not_empty(block_strs, "[br][br]")?
+            }
 
-        Text(text) => text.value.clone(),
+            // Paragraph: gather inline children as a single line.
+            Paragraph(par) => self.walk_inline_nodes(&par.children, level),
 
-        // TODO: more langs?
-        Code(md::Code { value, .. }) => format!("[codeblock]{value}[/codeblock]"),
+            // Inline code -> [code]...[/code]
+            InlineCode(md::InlineCode { value, .. }) => format!("[code]{value}[/code]"),
 
-        Paragraph(paragraph) => walk_nodes(&paragraph.children, definitions, ""),
+            // Strikethrough -> [s]...[/s]
+            Delete(del) => {
+                let inner = self.walk_inline_nodes(&del.children, level);
+                format!("[s]{inner}[/s]")
+            }
 
-        // BBCode supports lists, but docs don't.
-        List(_) | Blockquote(_) | FootnoteReference(_) | FootnoteDefinition(_) | Table(_) => {
-            String::new()
-        }
+            // Italic -> [i]...[/i]
+            Emphasis(emp) => {
+                let inner = self.walk_inline_nodes(&emp.children, level);
+                format!("[i]{inner}[/i]")
+            }
 
-        Html(html) => html.value.clone(),
+            // Bold -> [b]...[/b]
+            Strong(str) => {
+                let inner = self.walk_inline_nodes(&str.children, level);
+                format!("[b]{inner}[/b]")
+            }
 
-        _ => walk_nodes(node.children()?, definitions, ""),
-    };
+            // Plain text -> just the text, with newlines replaced by spaces.
+            Text(txt) => txt.value.replace("\n", " "),
 
-    Some(bbcode)
-}
+            // Heading -> single line, "fake" heading with [b]...[/b]
+            Heading(h) => {
+                let inner = self.walk_inline_nodes(&h.children, level);
+                format!("[b]{inner}[/b]")
+            }
 
-/// Calls [`walk_node`] over every node it receives, joining them with the supplied separator.
-fn walk_nodes(nodes: &[md::Node], definitions: &HashMap<&str, &str>, separator: &str) -> String {
-    nodes
-        .iter()
-        .filter_map(|n| walk_node(n, definitions))
-        .collect::<Vec<_>>()
-        .join(separator)
+            // Blockquote -> each child is effectively a block. We gather them with a single
+            // [br] in between, then prefix each resulting line with "> ".
+            Blockquote(bq) => {
+                let child_blocks: Vec<_> = bq
+                    .children
+                    .iter()
+                    .filter_map(|child| self.walk_node(child, level))
+                    .collect();
+                let content = child_blocks.join("[br]"); // each child is a block
+
+                // Prefix each line with "> "
```

Comment:
Nitpick: we typically treat comments as sentences (uppercase first word + period at end).
```suggestion
                // Prefix each line with "> ".
```

Please also adjust in other places :wink:

Reply:
Resolved.